src/ripple/app/tx/impl/XChainBridge.cpp

```diff
@@ -0,0 +1,1675 @@
+{
+    PaymentSandbox psb(&ctx_.view());
+
+    auto const account = ctx_.tx[sfAccount];
+    auto const amount = ctx_.tx[sfAmount];
+    auto const bridge = ctx_.tx[sfXChainBridge];
+
+    auto const sle = psb.peek(keylet::account(account));
+    if (!sle)
+        return tecINTERNAL;  // LCOV_EXCL_LINE
+
+    auto const sleB = psb.read(keylet::bridge(bridge));
+    if (!sleB)
+        return tecINTERNAL;  // LCOV_EXCL_LINE
+
+    auto const dst = (*sleB)[sfAccount];
+
+    auto const thTer = transferHelper(
+        psb,
+        account,
+        dst,
+        /*dstTag*/ std::nullopt,
+        /*claimOwner*/ std::nullopt,
+        amount,
+        TransferHelperCanCreateDst::no,
+        ctx_.journal);
+
+    if (!isTesSuccess(thTer))
+        return thTer;
+
+    psb.apply(ctx_.rawView());
+
+    return tesSUCCESS;
+}
+
+//------------------------------------------------------------------------------
+
+NotTEC
+XChainCreateClaimID::preflight(PreflightContext const& ctx)
+{
+    if (!ctx.rules.enabled(featureXChainBridge))
+        return temDISABLED;
+
+    if (auto const ret = preflight1(ctx); !isTesSuccess(ret))
+        return ret;  // LCOV_EXCL_LINE
+
+    if (ctx.tx.getFlags() & tfUniversalMask)
+        return temINVALID_FLAG;
+
+    auto const reward = ctx.tx[sfSignatureReward];
+
+    if (!isXRP(reward) || reward.signum() < 0 || !isLegalNet(reward))
+        return temXCHAIN_BRIDGE_BAD_REWARD_AMOUNT;
+
+    return preflight2(ctx);
+}
+
+TER
+XChainCreateClaimID::preclaim(PreclaimContext const& ctx)
+{
+    auto const account = ctx.tx[sfAccount];
+    auto const bridgeSpec = ctx.tx[sfXChainBridge];
+    auto const bridge = ctx.view.read(keylet::bridge(bridgeSpec));
+
+    if (!bridge)
+    {
+        return tecNO_ENTRY;
+    }
+
+    // Check that the reward matches
+    auto const reward = ctx.tx[sfSignatureReward];
+
+    if (reward != (*bridge)[sfSignatureReward])
+    {
+        return tecXCHAIN_REWARD_MISMATCH;
+    }
+
+    {
+        // Check reserve
+        auto const sle = ctx.view.read(keylet::account(account));
+        if (!sle)
+            return terNO_ACCOUNT;  // LCOV_EXCL_LINE
+
+        auto const balance = (*sle)[sfBalance];
+        auto const reserve =
+            ctx.view.fees().accountReserve((*sle)[sfOwnerCount] + 1);
+
+        if (balance < reserve)
+            return tecINSUFFICIENT_RESERVE;
+    }
+
+    return tesSUCCESS;
+}
+
+TER
+XChainCreateClaimID::doApply()
+{
+    auto const account = ctx_.tx[sfAccount];
+    auto const bridge = ctx_.tx[sfXChainBridge];
+    auto const reward = ctx_.tx[sfSignatureReward];
+    auto const otherChainSrc = ctx_.tx[sfOtherChainSource];
+
+    auto const sleAcc = ctx_.view().peek(keylet::account(account));
+    if (!sleAcc)
+        return tecINTERNAL;  // LCOV_EXCL_LINE
+
+    auto const sleB = ctx_.view().peek(keylet::bridge(bridge));
+    if (!sleB)
+        return tecINTERNAL;  // LCOV_EXCL_LINE
+
+    std::uint32_t const claimID = (*sleB)[sfXChainClaimID] + 1;
```

Comment:
You're adding one to the ClaimID before it is created.  This makes the first created ClaimID one, not zero.   In other parts of the XRP Ledger sequential IDs tend to start with zero, not one. Examples include (pre account delete) AccountRoot sequence numbers, NFT sequence IDs.

It would be nice if `XChainClaimID`s had a similar behavior to other parts of the ledger.

Reply:
Fixed in f2c40d3b5 [fold] Make the claim id start at zero, not one